### PR TITLE
Implement new validation spec for CustomProperties

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

File: SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/CustomProperties.cs [hunk 1]
```diff
@@ -109,31 +130,41 @@ public CustomProperties Clear(string key)
...
+                MobileCenterLog.Error(MobileCenterLog.LogTag, "Custom property \"" + key + "\" key length cannot be longer than \"" + MaxCustomPropertiesKeyLength + "\" characters.");
+                return false;
+            }
             if (Properties.ContainsKey(key))
             {
                 MobileCenterLog.Error(MobileCenterLog.LogTag, "Custom property \"" + key + "\" is already set or cleared and will be overridden.");
             }
+            else if (Properties.Count >= MaxCustomPropertiesCount)
+            {
+                MobileCenterLog.Error(MobileCenterLog.LogTag, "Custom properties cannot contain more than \"" + MaxCustomPropertiesCount + "\" items.");
```
Comment (guperrot) [inline SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/CustomProperties.cs]: Please remove quotes for numbers

File: SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/CustomProperties.cs [hunk 2]
```diff
@@ -109,31 +130,41 @@ public CustomProperties Clear(string key)
...
+        private bool ValidateKey(string key)
         {
             if (key == null || !KeyPattern.IsMatch(key))
             {
                 MobileCenterLog.Error(MobileCenterLog.LogTag, "Custom property \"" + key + "\" must match \"" + KeyPattern + "\"");
                 return false;
             }
+            if (key.Length > MaxCustomPropertiesKeyLength)
+            {
+                MobileCenterLog.Error(MobileCenterLog.LogTag, "Custom property \"" + key + "\" key length cannot be longer than \"" + MaxCustomPropertiesKeyLength + "\" characters.");
```
Comment (guperrot) [inline SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/CustomProperties.cs]: Please remove quotes for numbers

File: SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/CustomProperties.cs [hunk 3]
```diff
@@ -109,31 +130,41 @@ public CustomProperties Clear(string key)
...
         private CustomProperties SetObject(string key, object value)
         {
-            if (IsValidKey(key))
+            if (ValidateKey(key))
             {
-                if (value != null)
+                if (value == null)
                 {
-                    Properties[key] = value;
+                    MobileCenterLog.Error(MobileCenterLog.LogTag, "Custom property value cannot be null, did you mean to call clear?");
```
Comment (guperrot) [inline SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/CustomProperties.cs]: add key in variable message like Android https://github.com/Microsoft/mobile-center-sdk-android/pull/437/files#diff-9f25f34f214d1a0aa8070633251140a8R163

Comment (MatkovIvan) [inline SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/CustomProperties.cs]: @guperrot Done. But you link to other message. [Same message](https://github.com/Microsoft/mobile-center-sdk-android/pull/437/files#diff-9f25f34f214d1a0aa8070633251140a8R159) doesn't have key.

**Pull request merged**